### PR TITLE
Adam Goodchild

### PR DESCRIPTION
**Pull request by akorolczuk**
https://github.com/mimminito/TankBattalion

### CODE REVIEW

File: AdamGoodchild/Core/Health.cs [hunk 1]
```diff
@@ -0,0 +1,211 @@
+﻿using System.Collections;
```
Comment (akorolczuk) [inline AdamGoodchild/Core/Health.cs]: namespaces, regions, object pooling.. Starts promising ;)

**Comment by Moniraego** [comment]
I would say not great, but not bad. Given the time restriction and amount of work, a lot of the sins are forgivable. Nothing made me want to be a 'hard pass' here, but some criticisms:

- Very unity component oriented. Pretty well every class is a monobehaviour.
- Basic singleton pattern. This is the kind of thing that's excusable in such a small project though.
- No interfaces, or much architecture at all really. He wasn't trying to show off architecture I guess.
- Not very succinct
- A monobehaviour called LineOfSightShoot worries me.

That kind of thing. However I don't think this challenge is really 'show us your best work' because best work takes time, so hard to judge the code.

**Comment by akorolczuk** [comment]
Yeah I agree with @Moniraego. Test is not excellent but perhaps he'd have a good reasoning behind his choices. Time constraint forces you to cut some corners. If he proves his knowledge about OOP and production level game architecture he might be a good fit.

**Comment by akorolczuk** [comment]
So I'd say we should interview him.

**Pull request closed without merging**